### DISCUSSION TITLE
Country Data Component Question

### DISCUSSION THREAD

**Discussion by cuckoo-bird** [pimcore/pimcore#16990] [Q&A / Support]
Hi Guys,

I am a beginner and would like to ask how I can modify the data component functionality (ie extend the JS). I added one from the selection
<img width="559" alt="Screen Shot 2024-04-26 at 12 59 42 PM" src="https://github.com/pimcore/pimcore/assets/30544529/30a95741-e8c3-445e-8b99-4cc6b0ea7138">

and every time I search I get multiple suggestions. I would like to make the search start with the first letter. For example typing on search "Au" should suggest countries starting with "Au" and not anything with "au" in the middle or last of the string
<img width="442" alt="Screen Shot 2024-04-26 at 12 56 31 PM" src="https://github.com/pimcore/pimcore/assets/30544529/3fa08601-5d23-424c-995c-5ff8385795f9">

Thanks in advance
Ryan

**Comment by markus-moser** [comment]
The UI is based on Ext JS. It would be possible to override the field definition type for the country component and change the behaviour but this is nothing which I would really recommend. You could potentially get problems when updating Pimcore to newer versions (with the need to adapt your custom code after Pimcore updates). For your example the select box is created here:

https://github.com/pimcore/admin-ui-classic-bundle/blob/1.x/public/js/pimcore/object/classes/data/country.js#L126

So you would need to find a way in Ext JS to do what you would like to achieve and then override the country field type definition of the core in a bundle (https://pimcore.com/docs/platform/Pimcore/Extending_Pimcore/Bundle_Developers_Guide/Pimcore_Bundles/).
